$NetBSD: patch-aa,v 1.4 2014/12/15 05:52:01 mef Exp $

(from cvs log)
date: 2008-07-25 02:13:00 +0900;  author: tonnerre;  state: Exp;  lines: +23 -14;
Fix denial of sevice vulnerability in Berkeley yacc (CVE-2008-3196).

--- skeleton.c.orig	2005-05-05 01:39:36.000000000 +0200
+++ skeleton.c
@@ -87,6 +87,7 @@ char *header[] =
     "short   *yyssp;",
     "YYSTYPE *yyvsp;",
     "YYSTYPE  yyval;",
+    "static YYSTYPE yyvalzero;", /* no "const", must compile as C++ */
     "YYSTYPE  yylval;",
     "",
     "/* variables for the parser stack */",
@@ -275,7 +275,10 @@ char *body[] =
     "                YYPREFIX, yystate, yyn, yyrule[yyn]);",
     "#endif",
     "    yym = yylen[yyn];",
-    "    yyval = yyvsp[1-yym];",
+    "    if (yym)",
+    "        yyval = yyvsp[1-yym];",
+    "    else",
+    "        yyval = yyvalzero;",
     "    switch (yyn)",
     "    {",
     0
